beacon-chain/rpc/eth/beacon/pool.go

```diff
@@ -115,6 +115,16 @@ func (bs *Server) SubmitAttestations(ctx context.Context, req *ethpbv1.SubmitAtt
 		if err := bs.Broadcaster.BroadcastAttestation(ctx, subnet, att); err != nil {
 			broadcastFailed = true
 		}
+
+		if corehelpers.IsAggregated(att) {
+			if err := bs.AttestationsPool.SaveAggregatedAttestation(att); err != nil {
+				log.WithError(err).Error("could not save aggregated att")
+			}
+		} else {
+			if err := bs.AttestationsPool.SaveUnaggregatedAttestation(att); err != nil {
+				log.WithError(err).Error("could not save unaggregated att")
+			}
+		}
```

Comment:
We already save in the goroutine on line 95. But we assume it's unaggregated. We also do this assuming the attestation is unaggregated. I don't know why this assumption was made. This is probably the actual thing to fix.
```
bs.OperationNotifier.OperationFeed().Send(&feed.Event{
		Type: operation.UnaggregatedAttReceived,
		Data: &operation.UnAggregatedAttReceivedData{
			Attestation: att,
		},
})
```

Reply:
ah I missed that, I'll amend to that, thanks!